app/models/article.rb

```diff
@@ -14,6 +14,15 @@ def self.create_articles(articles_data)
     end
   end
 
+  def keyword_base_text
+    base_text_data = [title,
+                      desc_facet.gsub(/,/, " "),
+                      abstract]
+    base_text_data.map do |attribute|
+      attribute.gsub(/[^\w]+/, " ").split(" ")
```

Comment:
I'm basing the keywords of an article on its title, desc_facet, and abstract.  The first regex just gets rid of any commas in the desc_facet.  The second regex replaces any non-word characters with a space.